pkg/kubelet/kubelet_pods.go

```diff
@@ -135,7 +136,37 @@ func makeMounts(pod *v1.Pod, podDir string, container *v1.Container, hostName, h
 			return nil, err
 		}
 		if mount.SubPath != "" {
+			fileinfo, err := os.Lstat(hostPath)
+			if err != nil {
+				return nil, err
+			}
+			gid := fileinfo.Sys().(*syscall.Stat_t).Gid
+			perm := fileinfo.Mode()
+
 			hostPath = filepath.Join(hostPath, mount.SubPath)
+
+			// If fsGroup volume ownership was set for the mounter path, set it for the sub path too
+			fsGroup := pod.Spec.SecurityContext.FSGroup
+			if gid == uint32(*fsGroup) {
+				if err := os.MkdirAll(hostPath, perm); err != nil {
+					glog.Errorf("failed to mkdir:%s", hostPath)
+					return nil, err
+				}
+				volume.SetVolumeOwnership(vol.Mounter, fsGroup, hostPath)
+
+				// If umask is preventing us from making the sub path have the same permissions as the
+				// mounter path, chmod the sub path
+				fileinfo, err := os.Lstat(hostPath)
```

Comment:
I actually just convinced myself that it's SetVolumeOwnership that isn't needed here. :/ Because the sub path will inherit the fsGroup owner and setgid bit from the parent mount path if the parent mount path had SetVolumeOwnership called on it already... But, regardless of what was done to the parent mount path & how, all I need to do is make the sub path have the same permissions as it, so I should just chown it + chmod it to look exactly like the mount path right? There's no need to do any gid owner checks, just chown & chmod the sub path to look like the mount path because that's what the user expects.